spec/models/book_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'rails_helper'
+
+describe Book do
+  subject(:book) { build(:book) }
+  subject(:invalid_year_book) { build(:book, year: '125.4') }
+  subject(:missing_author_book) { build(:book, author: nil) }
+  subject(:empty_title_book) { build(:book, title: '') }
+  subject(:empty_genre_book) { build(:book, genre: '') }
+  subject(:empty_publisher_book) { build(:book, publisher: '') }
+  subject(:missing_image_book) { build(:book, image: nil) }
+
+  describe '#save' do
+    context 'When the book is created' do
+      it 'generates an id' do
+        book.save!
+        expect(book.id).to be_present
+      end
+    end
+
+    context 'when year is invalid' do
+      it 'fails at save' do
+        expect { invalid_year_book.save! }.to raise_error(ActiveRecord::RecordInvalid)
```

Comment:
You also should test error message [check this](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/raise-error-matcher)
```rb
it 'raise RecordInvalid error' do
   error_msg = 'Error message' # Remplace ´Error message´ with correct error message
   expect { invalid_year_book.save! }.to raise_error(ActiveRecord::RecordInvalid, error_msg)
end
```